search/views.py

```diff
@@ -1,3 +1,121 @@
-from django.shortcuts import render
+import re
 
-# Create your views here.
+from decimal import Decimal
+
+from django.http.response import JsonResponse
+from django.views         import View
+from django.db.models     import Q
+
+from .models       import Amenity
+from homes.models  import DongType, GuType
+from .get_response import get_response
+from mysettings    import DEFAULT_MAP_POINT, MEAN_EARTH_RADIUS
+class MapView(View):
+    def get(self, request):
+        try:
+            if not request.GET.get('center'):
+                latitude    = DEFAULT_MAP_POINT['center']["latitude"]
+                longitude   = DEFAULT_MAP_POINT['center']["longitude"]
+                zoom        = DEFAULT_MAP_POINT["zoom"]
+                q           = Q()
+                
+                return JsonResponse(get_response(latitude, longitude, zoom, q), status = 200)
+            
+            coords           = request.GET.get('center')
+            zoom             = int(request.GET.get('zoom'))
+            room_type        = request.GET.get('room_type')
+            sale_type        = request.GET.get('sale_type')
+            deposit          = request.GET.get('deposit')
+            monthly_pay      = request.GET.get('monthly_pay')
+            maintenance_cost = request.GET.get('maintenance_cost')
+            exclusive_m2     = request.GET.get('exclusive_m2')
+            
+            if coords.find(',') != -1:
+                lat, lng  = coords.split(',')
+                latitude  = Decimal(lat)
+                longitude = Decimal(lng)
+            
+            q = Q()
+            if room_type:
+                room_types  = room_type.split(',')
+                for condition in room_types:
+                    q |= Q(room_type__name = condition)
```

Comment:
```suggestion
            if room_type:
                room_types  = room_type.split(',')
                q |= Q(room_type__name__in = room_type)
```
list를 한 번에 조건에 넣을 수 있는 방법입니다. 이 방법 참고해서 수정 가능한 부분들 수정해 주세요

Reply:
Q안에 배열을 넣어도 적용된단 말씀이시죠?? 쉘에서 테스트해보겠습니다~!

Reply:
Q안에 or조건으로 넣는 방법을 못찾았습니다!

Reply:
수정 완료!